lib/guest_configuration_base.pm

```diff
@@ -0,0 +1,1408 @@
+        my $_guest_name_in_metadata = grep { /name=(.*)/ } split(',', $self->{guest_metadata});
+        if ($self->{guest_name} ne $_guest_name_in_metadata) {
+            diag("The guest instance has a different name $_guest_name_in_metadata in metadata. Reset guest name to it.");
+            $self->{guest_name} = $_guest_name_in_metadata;
+        }
+    }
+    if ($self->{guest_version} eq lc get_required_var('VERSION')) {
+        $self->{guest_build} = lc get_required_var('BUILD');
+    }
+    else {
+        $self->{guest_build} = 'GM';
+    }
+    return $self;
+}
+
+#Print out guest instance parameters
+sub print_guest_params {
+    my $self = shift;
+
+    $self->reveal_myself;
+    diag("The guest instance $self->{guest_name} created is as below:");
+    foreach (keys %{$self}) {
+        if (defined $self->{$_}) {
+            if (ref($self->{$_}) eq 'HASH') {
+                diag $_ . ' => ' . Dumper(\%{$self->{$_}});
+            }
+            elsif (ref($self->{$_}) eq 'ARRAY') {
+                diag $_ . ' => ' . @{$self->{$_}};
+            }
+            else {
+                diag $_ . ' => ' . $self->{$_};
+            }
+        }
+        else {
+            diag $_ . ' => ' . 'undef';
+        }
+    }
+    return $self;
+}
+
+#Install necessary packages, patterns, create [common_log_folder].These are common environment which will be used for all guest instances.
+sub prepare_common_environment {
+    my $self = shift;
+
+    $self->reveal_myself;
+    if ($common_environment_prepared eq 'false') {
+        $self->clean_up_all_guests;
+        script_run("rm -f -r $common_log_folder");
+        assert_script_run("mkdir -p $common_log_folder");
+        my $_packages_to_check = 'wget curl screen dnsmasq xmlstarlet yast2-schema python3 nmap';
+        zypper_call("install -y $_packages_to_check");
+        my $_patterns_to_check = 'kvm_server kvm_tools';
+        $_patterns_to_check = 'xen_server xen_tools' if ($self->{host_virt_type} eq 'xen');
+        zypper_call("install -y -t pattern $_patterns_to_check");
+        $common_environment_prepared = 'true';
+        diag("Common environment preparation is done now.");
+    }
+    else {
+        diag("Common environment preparation had already been done.");
+    }
+    return $self;
+}
+
+#Remove all existing guests and storage files in /var/lib/libvirt/images/
+sub clean_up_all_guests {
+    my $self = shift;
+
+    my @_guests_to_clean_up = split(/\n/, script_output("virsh list --all --name", proceed_on_failure => 1));
+    if (scalar(@_guests_to_clean_up) gt 0) {
+        diag("Going to clean up all guests on $self->{host_name}");
+        foreach (@_guests_to_clean_up) {
+            script_run("virsh destroy $_");
+            script_run("virsh undefine $_ --nvram") if (script_run("virsh undefine $_") ne 0);
+        }
+        script_run("rm -f -r /var/lib/libvirt/images/*");
```

Comment:
Regarding the storage directory, suggest to make it configurable and refer to the configured value, rather than hard code(thru all the files)